src/functional/ffront/past_to_itir.py

```diff
@@ -153,20 +153,21 @@ def _visit_slice_bound(
             raise AssertionError("Expected `None` or `past.Constant`.")
         return lowered_bound
 
-    def _construct_itir_out_arg(self, node: past.TupleExpr | past.Name) -> itir.Expr:
+    def _construct_itir_out_arg(
+        self, node: past.TupleExpr | past.Name | past.Subscript
+    ) -> itir.Expr:
         if isinstance(node, past.Name):
             return itir.SymRef(id=node.id)
+        elif isinstance(node, past.Subscript):
+            return self._construct_itir_out_arg(node.value)
         elif isinstance(node, past.TupleExpr):
-            if any(isinstance(el, past.Subscript) for el in _flatten_tuple_expr(node)):
-                raise NotImplementedError("Slicing for tuple `out` argument not supported.")
             return itir.FunCall(
                 fun=itir.SymRef(id="make_tuple"),
                 args=[self._construct_itir_out_arg(el) for el in node.elts],
             )
         else:
             raise AssertionError()(
-                "Unexpected `out` argument. Must be a `past.Name` or nesting of "
-                "`past.TupleExpr` and `past.Name`."
+                "Unexpected `out` argument. Must be a `past.Name` or `past.Subscript` or nesting in a `past.TupleExpr`."
```

Comment:
Not quite sure I understand this error message. What does it mean for the `out` argument to be a "nesting in a past.TupleExpr"?

Reply:
I'll make it more verbose. It means it can be a TupleExpr which contains either Name, Subscript or another TupleExpr which contains either Name, Subscript or another TupleExpr, ... recurse